src/QMCWaveFunctions/BsplineFactory/OffloadMultiWalkerMem.h

```diff
@@ -0,0 +1,51 @@
+//////////////////////////////////////////////////////////////////////////////////////
+// This file is distributed under the University of Illinois/NCSA Open Source License.
+// See LICENSE file in top directory for details.
+//
+// Copyright (c) 2021 QMCPACK developers.
+//
+// File developed by: Ye Luo, yeluo@anl.gov, Argonne National Laboratory
+//
+// File created by: Ye Luo, yeluo@anl.gov, Argonne National Laboratory
+//////////////////////////////////////////////////////////////////////////////////////
+
+
+#ifndef QMCPLUSPLUS_OFFLOADSHAREDMEM_H
+#define QMCPLUSPLUS_OFFLOADSHAREDMEM_H
+
+#include "OMPTarget/OMPallocator.hpp"
+#include "PinnedAllocator.h"
+#include "ResourceCollection.h"
+
+namespace qmcplusplus
+{
+
+template<typename ST, typename TT>
+struct OffloadMultiWalkerMem: public Resource
+{
+  template<typename DT>
+  using OffloadPinnedAllocator = OMPallocator<DT, PinnedAlignedAllocator<DT>>;
+
```

Comment:
If I am a scope takes an OffloadMultiWalkerMem as an argument
what can I assume about state of the OMWM.

I'd assume mw_ratios_private, rg_private, mw_pos_copy might contain persistent data written to by some scopes maybe read by others. 

I'd assume mw_offload_scratch, mw_results_scratch, buffer_H2D I can just use for whatever I want to write transfer H2D read/write some more and then transfer D2H read.

det_ratios_buffer_H2D?